source/game/scene/GameScene.cc

```diff
@@ -81,4 +109,89 @@ void GameScene::unmountResources() {
         delete resource;
     }
 }
+
+void GameScene::checkMemory() {
+    EGG::ExpHeap *heap = EGG::Heap::dynamicCastToExp(m_heap);
+    ASSERT(heap);
+
+    heap->calcGroupSize(&m_groupSizeRecord);
+    size_t defaultSize = m_groupSizeRecord.getGroupSize(static_cast<u16>(GroupID::None));
+
+    // Because we're working with a class that can be inherited (but not doubly-inherited),
+    // it's possible that derived classes can have a different size compared to GameScene
+    // We don't need to know the exact derived class, though, as we have the memory block head
+    // The scene is the first, always group ID 0 allocation to happen in the scene's heap
+    Abstract::Memory::MEMiExpBlockHead *blockHead =
+            static_cast<Abstract::Memory::MEMiExpBlockHead *>(
+                    SubOffset(this, sizeof(Abstract::Memory::MEMiExpBlockHead)));
+    size_t initialAllocSize = blockHead->m_size;
+    ASSERT(defaultSize >= initialAllocSize);
+
+    defaultSize -= initialAllocSize;
+    if (defaultSize > 0) {
+        WARN("Default memory usage found! %zu bytes", defaultSize);
+    }
+
+    for (u16 groupID = 1; groupID < m_groupSizeRecord.size(); ++groupID) {
+        size_t size = m_groupSizeRecord.getGroupSize(groupID);
+        if (size == 0) {
+            continue;
+        }
+
+        DEBUG("Group ID %d: Allocated %zu bytes", groupID, size);
+    }
+}
+
+#ifdef BUILD_DEBUG
+void GameScene::getMemoryLeakTags() {
+    size_t count = getMemoryLeakTagCount();
+    if (count == 0) {
+        return;
+    }
+
+    std::span<u32> tags = std::span<u32>(new u32[count], count);
+    for (auto &tag : tags) {
+        tag = 0; // empty tag
+    }
+
+    EGG::Heap::dynamicCastToExp(m_heap)->dynamicCastHandleToExp()->visitAllocated(ViewTags,
+            GetAddrNum(&tags));
+
+    DEBUG("TAGGED MEMORY BLOCKS:");
+    printf("[%d", tags[0]);
+    for (u32 i = 1; i < tags.size(); ++i) {
+        printf(", %d", tags[i]);
+    }
+    printf("]\n");
+
+    delete[] tags.data();
+}
+
+size_t GameScene::getMemoryLeakTagCount() {
+    size_t count = 0;
+    EGG::Heap::dynamicCastToExp(m_heap)->dynamicCastHandleToExp()->visitAllocated(IncreaseTagCount,
+            GetAddrNum(&count));
+    return count;
+}
+
+void GameScene::ViewTags(void *block, Abstract::Memory::MEMiHeapHead * /*heap*/, uintptr_t param) {
+    Abstract::Memory::MEMiExpBlockHead *blockHead =
+            static_cast<Abstract::Memory::MEMiExpBlockHead *>(
+                    SubOffset(block, sizeof(Abstract::Memory::MEMiExpBlockHead)));
+    std::span<u32> *span = reinterpret_cast<std::span<u32> *>(param);
+    for (auto &tag : *span) {
+        if (tag == 0) {
+            tag = blockHead->m_tag;
+            break;
+        }
+    }
+}
+
+void GameScene::IncreaseTagCount(void * /*block*/, Abstract::Memory::MEMiHeapHead * /*heap*/,
+        uintptr_t param) {
+    size_t &count = *reinterpret_cast<size_t *>(param);
+    ++count;
+}
+#endif
```

Comment:
We should use the same convention we use for namespace end braces.
```suggestion
#endif // BUILD_DEBUG
```